### PR TITLE
Feature allowing recurring payments to be scheduled for the last day of the month

### PR DESCRIPTION
**Pull request by kg8z**
Issue: #2039 

## PR Type
What kind of change does this PR introduce?

- Feature

## What is the current behavior?
Recurring payments can only be scheduled for specific dates. The user cannot specify recurring payments that occur on the last day of the month.

Logic to determine when new transactions should be created for quarterly and biannual recurrence types is based on days rather than months, which is imprecise.

Logic to determine the payment date for a new recurring payment transaction uses DateTime.Today for every recurrence type except monthly. This causes the day of month to change for bimonthly, quarterly, biannually, and yearly recurrence types.

The "Enddate" string resource value is missing a space.

## What is the new behavior?
Recurring payments now have a "Last Day of Month" option. If selected, new recurring payment transactions will be created on the last day of the month. _Note: This does not affect the payment date of the original payment entered by the user, only transactions that are automatically created when the recurrence period has elapsed._

Refactored the recurring payment logic used to determine when a new transaction should be created. Quarterly and biannual recurrence types are now based on months (3 and 6, respectively) rather than days.

Refactored the logic to determine the payment date for new recurring payment transactions. Daily, daily without weekend, weekly, and biweekly still use DateTime.Today. Monthly, bimonthly, quarterly, biannually, and yearly now use the day of month from the original payment date to determine the new transaction date, keeping the day of month consistent for these recurrence types.

The "Enddate" string resource value has been changed to "End Date".

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code on Windows
- [x] Tested code on Android
- [x] Tested code on iOS
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes

### CODE REVIEW

File: Src/MoneyFox.Win/MoneyFox.Win/Converter/LastDayOfMonthVisibilityConverter.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+namespace MoneyFox.Win.Converter;
+
+using Core.Aggregates.Payments;
+using Core._Pending_.Common.Helpers;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using System;
+
+public class LastDayOfMonthVisibilityConverter : IValueConverter
```
Comment (NPadrutt) [inline Src/MoneyFox.Win/MoneyFox.Win/Converter/LastDayOfMonthVisibilityConverter.cs:9]: I think I would rather make a bool property in the viewmodel for that and bind the visibility to that.

File: Src/MoneyFox/Converter/LastDayOfMonthVisibilityConverter.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+namespace MoneyFox.Converter
+{
+
+    using System;
+    using Xamarin.Forms;
+    using System.Globalization;
+    using Core._Pending_.Common.Helpers;
+    using Core.Aggregates.Payments;
+
+    public class LastDayOfMonthVisibilityConverter : IValueConverter
```
Comment (NPadrutt) [inline Src/MoneyFox/Converter/LastDayOfMonthVisibilityConverter.cs:10]: Same here as mentioned above.

**Review by NPadrutt — CHANGES_REQUESTED** [review]
Looks good. Only had two small comments. :)

File: Src/MoneyFox.Win/MoneyFox.Win/Converter/LastDayOfMonthVisibilityConverter.cs (hunk 1 above)
Comment (kg8z) [reply Src/MoneyFox.Win/MoneyFox.Win/Converter/LastDayOfMonthVisibilityConverter.cs]: @NPadrutt - That's an excellent idea. I removed the stand-alone iValueConverter class and moved the visibility logic into the ViewModel. Much cleaner. Thanks for the tip!

File: Src/MoneyFox/Converter/LastDayOfMonthVisibilityConverter.cs (hunk 2 above)
Comment (kg8z) [reply Src/MoneyFox/Converter/LastDayOfMonthVisibilityConverter.cs]: Same comment as above.

**Comment by NPadrutt** [comment]
thanks @kg8z  :)

**Pull request merged**